Jenkinsfile

```diff
@@ -15,6 +9,23 @@ pipeline {
                 }
             }
             stages {
+                stage('set_version') {
+                    steps {
+                        sh 'bumpversion patch'
+                    }
+                }
+                stage('release') {
+                    when { branch 'master' }
+                    steps {
+                        withCredentials([usernamePassword(credentialsId: env.CREDENTIALS_ID, passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                            sh '''
+                                export VERSION=$(bump2version --list --allow-dirty release | grep new_version= | sed -r s,"^.*=",,)
+                                git push origin master
+                                git push origin refs/tags/v$VERSION
```

Comment:
Can be written in a cleaner CLI:
`git push origin tag v$VERSION`

from git-push man page:
```tag <tag> means the same as refs/tags/<tag>:refs/tags/<tag>.```